prql-compiler/src/semantic/std.prql

```diff
@@ -33,17 +33,16 @@ func aggregate<table> a<column> tbl<table> -> null
 func sort<table> by tbl<table> -> null
 func take<table> expr tbl<table> -> null
 func join<table> `default_db.with`<table> filter `noresolve.side`:inner tbl<table> -> null
-func append<table> `default_db.bottom`<table> top<table> -> null
-func union<table> `default_db.bottom`<table> top<table> -> (
-    from t = _param.top | append _param.bottom | group [t.*] (take 1)
-)
 func group<table> by pipeline tbl<table> -> null
 func window<table> rows:0..0 range:0..0 expanding:false rolling:0 pipeline tbl<table> -> null
 
-func intersection<table> bottom<table> top<table> -> (
-    from t = _param.top | join b = _param.bottom (all (map _eq (zip t.* b.*))) | select t.*
+func append<table> `default_db.bottom`<table> top<table> -> null
+func intersect<table> `default_db.bottom`<table> top<table> -> (
+    from t = _param.top
+    join b = _param.bottom (all (map _eq (zip t.* b.*)))
```

Comment:
What's this? :)

Reply:
What do you mean? the `(all (map _eq (zip t.* b.*)))` part?

`t.*` and `b.*` each evaluate to a list of columns.

`(zip t.* b.*)` evaluates to `[[t.col1, b.col1], [t.col2, b.col2], ...]`

`(map _eq (zip t.* b.*))` evaluates to `[t.col1 == b.col1, t.col2 == b.col2, ...]`

`(all (map _eq (zip t.* b.*)))` evaluates to `(t.col1 == b.col1 and t.col2 == b.col2) and ...`

... it's a join condition over all columns of both tables, abusing our resolver.

The helper functions can remain an implementation details. It'd be nice if we could hide them from the public interface.